x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/CacheFile.java

```diff
@@ -85,7 +85,8 @@ public Path getFile() {
 
     ReleasableLock fileLock() {
         boolean success = false;
-        final ReleasableLock fileLock = readLock.acquire();
+        final ReleasableLock fileLock = new ReleasableLock(readLock);
```

Comment:
Can we return `fileLock::close` if we make this method return `Releasable` instead?

Reply:
Sure, in that case we don't need the `ReleasableLock` at all and can return `readLock::unlock`: I pushed ebae53d76e70a2b3282273377e4c85c26f06b301